### PR TITLE
cleanup issueCredentialOptions and VerifyOptions

### PR DESCRIPTION
**Pull request by mkhraisha**
Forked off #373 
tackles #318 

Merge conflicts will be cleaned up once  once #373 is merged in.

Challenge changes are not intended to be reviewed in this PR, but are remnants from forking off #373. 

1. Updates `IssueCredentialOptions` to remove `created`, and updates the example
2. Updates `DeriveCredentialOptions` to add `selectivePointers`
3. Removes `VerifyOptions` this has now been split into:
4.  `VerifyCredentialOptions`: this currently only has `returnCredential` which is based on what the current spec shows.
5. `VerifyPresentationOptions`: this currently has the properties `domain` and `challenge`. The issue mentions adding `proofPurpose`, but I wasn't sure, iirc the `proofPurpose` of a presentation is always `authentication`. if that is not true happy to update the PR. The Spec also mentions `returnCredential` as a possible option to this endpoint, I'm not sure if thats an error that should be `returnPresentation` or it should just be excluded entirely, I erred on the side of exclusion.
6.  `CreatePresentationOptions`:  intended to provide options when creating a presentation, similar question on the `proofPurpose` field.
7. Updates references that call `VerifyOptions`.

***
<a href="https://pr-preview.s3.amazonaws.com/w3c-ccg/vc-api/pull/375.html" title="Last updated on Mar 5, #2024, #9:44 PM UTC (ebd12e1)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c-ccg/vc-api/375/8a94c1f...ebd12e1.html" title="Last updated on Mar 5, #2024, #9:44 PM UTC (ebd12e1)">Diff</a>
    Don't remove this comment or modify anything below this line.
    If you don't want a preview generated for this pull request,
    just replace the whole of this comment's content by "no preview"
    and remove what's below.
-->
***
<a href="https://pr-preview.s3.amazonaws.com/w3c-ccg/vc-api/pull/375.html" title="Last updated on Mar 26, 2024, 7:11 PM UTC (7fa06a1)">Preview</a> (<a href="https://pr-preview.s3.amazonaws.com/w3c-ccg/vc-api/pull/375.html#Don't remove this comment or modify anything below this line.
    If you don't want a preview generated for this pull request" title="#Don't remove this comment or modify anything below this line.
    If you don't want a preview generated for this pull request">#Don't…</a>) (<a href="https://pr-preview.s3.amazonaws.com/w3c-ccg/vc-api/pull/375.html#just replace the whole of this comment's content by "no preview"
    and remove what's below.

-->
***
<a href="https://pr-preview.s3.amazonaws.com/w3c-ccg/vc-api/pull/375.html" title="Last updated on Mar 5" title="#just replace the whole of this comment's content by "no preview"
    and remove what's below.

-->
***
<a href="https://pr-preview.s3.amazonaws.com/w3c-ccg/vc-api/pull/375.html" title="Last updated on Mar 5">#just…</a>) (<a href="https://pr-preview.s3.amazonaws.com/w3c-ccg/vc-api/pull/375.html#2024" title="#2024">#2024</a>) (<a href="https://pr-preview.s3.amazonaws.com/w3c-ccg/vc-api/pull/375.html#9:44 PM UTC (ebd12e1)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c-ccg/vc-api/375/8a94c1f...ebd12e1.html" title="Last updated on Mar 5" title="#9:44 PM UTC (ebd12e1)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c-ccg/vc-api/375/8a94c1f...ebd12e1.html" title="Last updated on Mar 5">#9:44…</a>) (<a href="https://pr-preview.s3.amazonaws.com/w3c-ccg/vc-api/pull/375.html#2024" title="#2024">#2024</a>) (<a href="https://pr-preview.s3.amazonaws.com/w3c-ccg/vc-api/pull/375.html#9:44 PM UTC (ebd12e1)">Diff</a>" title="#9:44 PM UTC (ebd12e1)">Diff</a>">#9:44…</a>) | <a href="https://pr-preview.s3.amazonaws.com/w3c-ccg/vc-api/375/15a3e9b...7fa06a1.html" title="Last updated on Mar 26, 2024, 7:11 PM UTC (7fa06a1)">Diff</a>

### CODE REVIEW

File: components/DeriveCredentialOptions.yml
Comment (dlongley) [inline components/DeriveCredentialOptions.yml:31]:
```suggestion
          "selectivePointers": ["/credentialSubject/birthCountry"]
```

File: components/IssueCredentialOptions.yml
Comment (dlongley) [inline components/IssueCredentialOptions.yml:31]:
```suggestion
          "mandatoryPointers": ["/issuer", "/validFrom", "/validUntil"]
```

File: components/CreatePresentationOptions.yml
Comment (TallTed) [inline components/CreatePresentationOptions.yml:30]:
```suggestion
          description: A challenge provided by the requesting party of the proof. For example, 6e62f66e-67de-11eb-b490-ef3eeefa55f2.
        domain:
          type: string
          description: The intended domain of validity for the proof. For example, website.example.
```

**Review by msporny — APPROVED** [review]

**Comment by msporny** [comment]
@mkhraisha the group reviewed this today, looks like it's on a good path, we're getting ready to merge 373, please re-base once that happens and we'll merge this one as well. Thank you for the PR!

**Review by dlongley — APPROVED** [review]

**Comment by msporny** [comment]
Re-ping @mkhraisha, please re-base, we'd like to merge your PR.

**Comment by mkhraisha** [comment]
Rebased, hopefully this works now.

File: components/Challenge.yml
Comment (msporny) [inline components/Challenge.yml:67]:
```suggestion
```

File: verifier.yml
Comment (msporny) [inline verifier.yml:71]:
```suggestion
```

**Pull request merged**
